docs/index.md

```diff
@@ -6,69 +6,87 @@ This page is the top level entry point into the Substrate documentation.
 
 > **Some Expertise Needed**
 >
->In order to get the most out of Substrate, you should have a good knowledge of blockchain concepts
->and basic cryptography. Terminology like header, block, client, hash, transaction and signature
->should be familiar. At present you will need a working knowledge of Rust to be able to do any
->significant customization/adaption of Substrate (though eventually, we aim for this not to be the
->case).
+> In order to get the most out of Substrate, you should have a good knowledge of blockchain concepts and basic
+> cryptography. Terminology like header, block, client, hash, transaction and signature should be familiar. At present
+> you will need a working knowledge of Rust to be able to do any significant customization/adaption of Substrate (though
+> eventually, we aim for this not to be the case).
 
 Substrate is a blockchain development framework with a completely generic State Transition Function
-([STF](knowledgebase/getting-started/glossary#stf-state-transition-function)) and modular components for consensus, networking,
-and configuration.
-
-Despite being "completely generic", it comes with both standards and conventions - particularly with
-the Substrate runtime module library (a.k.a [FRAME](knowledgebase/runtime/frame.md)) - regarding the underlying
-data-structures that power the STF, thereby making rapid blockchain development a reality.
+([STF](knowledgebase/getting-started/glossary#stf-state-transition-function)) and modular components for consensus,
+networking, and configuration.
+
+Despite being "completely generic", it comes with both standards and conventions - particularly with the Substrate
+runtime module library (a.k.a [FRAME](knowledgebase/runtime/frame.md)) - regarding the underlying data-structures that
+power the STF, thereby making rapid blockchain development a reality.
+
+## Architecture
+
+![Substrate Client Architecture](assets/substrate-arch.png)
+
+The Substrate client is an application that runs a Substrate-based blockchain node - it consists of several components
+that include, but are not limited to, the following:
+
+- **Storage** is used to persist the evolving state of the decentralized system represented by a blockchain. The
+  blockchain network allows participants to reach trustless [consensus](knowledgebase/advanced/consensus) about the
+  state of storage. Substrate ships with a simple and highly efficient
+  [key-value storage mechanism](knowledgebase/advanced/storage).
+- **Runtime** logic defines how blocks are processed, including state transition logic. In Substrate, runtime code is
+  compiled to [Wasm](knowledgebase/getting-started/glossary#webassembly-wasm) and becomes part of the blockchain's
+  storage state - this enables one of the defining features of a Substrate-based blockchain:
+  [forkless runtime upgrades](knowledgebase/advanced/executor#forkless-runtime-upgrades). Substrate clients may also
```

Comment:
I dont believe this is the proper way to do Docusarus links

https://docusaurus.io/docs/en/navigation

You should target the `md` file which will then be processed by docusaurus to generate the correct url. Versus here, you are assuming how docusaurus will generate the link, and this will fail for things like versioning or whatever other fringe scenarios the documentation has.

Reply:
Probably true, but I'm going to stick with the existing conventions for now. We can make a separate sweep of internal links if we deem that worthwhile.